src/components/dls/ModalNew/Modal.tsx

```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import * as DialogPrimitive from '@radix-ui/react-dialog';
+import styles from './Modal.module.scss';
+
+const Modal = ({ children, trigger }) => (
+  <DialogPrimitive.Root>
+    <DialogPrimitive.Overlay className={styles.overlay} />
+    {trigger}
```

Comment:
Can we change this to have `<DialogPrimitive.Trigger>{trigger}</DialogPrimitive.Trigger>` instead of having to use `Modal.Trigger` as part of the prop every time `Modal` is being used?

Reply:
yes, I am still thinking a little bit about this. 
But one of the idea is that ideally, we should be able to use `<Button>` as trigger, so we can use buttons's props. such as `type` `variant`, `shape` etc. For example

```
<Modal trigger={<Button type={ButtonType.Secondary} variant={ButtonVariant.Ghost} />}>
```

I'll probably try to play a little bit if it's possible to do that.

do you have any comment/idea about this?

Reply:
I think we are forced to use `DialogPrimitive.Trigger` for any type. So we can introduce 2 new props for example:
1. `triggerType` which can be an enum called `TriggerTypes` to have `Button`, `Default` or others values  and set to `TriggerTypes.Default` by default.
2. `triggerProps` which is optional and can be an object and in the case of a `Button`, we can pass something like `{type:ButtonType.Secondary, variant:ButtonVariant.Ghost}`.
and we use it like this:

```
<DialogPrimitive.Trigger>
{triggerType === TriggerTypes.Button ? <Button {...triggerProps}>{trigger}</Button>: <div {...triggerProps}>{trigger}</div>}
</DialogPrimitive.Trigger>
```

What do you think?